### ISSUE TITLE
【bug】panic: duplicate metrics collector registration attempted

### ISSUE THREAD

**Issue by liguozhong** [grafana/loki#5129]
panic: duplicate metrics collector registration attempted

goroutine 1 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0x2e03640, {0xc000a15570, 0x1, 0x0})
	/Users/fuling/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/registry.go:403 +0x7f
github.com/prometheus/client_golang/prometheus/promauto.Factory.NewGauge({{0x342a890, 0xc000134a00}}, {{0x2e03640, 0x6}, {0x0, 0x0}, {0x2e4253a, 0x1c}, {0x2e6b779, 0x27}, ...})
	/Users/fuling/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/promauto/auto.go:297 +0x102
github.com/prometheus/client_golang/prometheus/promauto.NewGauge(...)
	/Users/fuling/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/promauto/auto.go:191
github.com/grafana/loki/pkg/distributor/clientpool.init()
	/Users/fuling/go/pkg/mod/github.com/grafana/loki@v1.6.2-0.20220112180552-34a4b242bf1d/pkg/distributor/clientpool/ingester_client_pool.go:14 +0xeb
	
![image](https://user-images.githubusercontent.com/9583245/149295160-ba544878-77e9-4904-aa9b-ce3963b95363.png)

**Comment by chaudum** [comment]
Hey @liguozhong could you provide some instructions how to reproduce the issue? Also which version (commit hash) you were using? Thanks!

**Comment by chaudum** [comment]
@liguozhong I this still an issue?

**Comment by liguozhong** [comment]
> @liguozhong I this still an issue?

@chaudum  thanks for patience！

This error occurs in a small probability scenario. Program projects depend on each other.

When one of my test demo expects to write loki, in order to reuse the client of promtail, an error is reported when loki is referenced through mod dependencies. Just as loki references thanos, we want to reuse existing code.

my test demo also depend the cortex program and hopes to reuse many of cortex's existing software.

suggestion：
If the go code of promtail can be extracted separately, it is the best for a new project, provide go sdk for writing loki。。。

**Comment by ldb** [comment]
I am experiencing the same problem.
Some information:
Versions tried: `main-651ac09` and `main-6c84304` (both newer than 2.4.2 but not yet released)

It always happened for me when promtail was configured with a `loki_push_api` scrape job. 
We have 2 scrape jobs that run fine on 2.4.2, but as soon as I updated to a more recent version, we encountered this error with a slightly different error message.

Things we tried:
- In the global `server` block, setting `register_instrumentation: false`
- In `loki_push_api.server` block, setting `register_instrumentation: false`
- Both of the above

All had the same result: Promtail crashing on startup.
The only way around this was to remove the `loki_push_api` scrape jobs which for us is not an option.
